error parsing config: repo config not allowed to set 'workflow' key: server-side config needs 'allowed_overrides: [workflow]'

I was reading the code here:

https://github.com/cresta/atlantis-drift-detection/blob/52efcd54d3b58dd9b0e9ef9232edd7b83fa287e6/internal/atlantis/config.go#L34

it looks like the action is parsing the repo side config `atlantis.yaml` and trying to find something that is `not` defined in there but instead in the `repos.yaml` config on the server side.

https://www.runatlantis.io/docs/custom-workflows.html#usage

`allowed_overrides: [workflow]` is a config on the server side that will not be on the monorepo where the terraform code resides.

the error I'm getting :

```
{"level":"info","ts":1672897241.4019797,"caller":"atlantis-drift-detection/main.go:108","msg":"setting up workflow notification"}
{"level":"debug","ts":1672897241.4021816,"caller":"atlantis-drift-detection/main.go:67","msg":"call call git clone ***github.com/jamengual/atlantmos.git /tmp/gogit2211482022"}
{"level":"debug","ts":1672897241.7879[17](https://github.com/jamengual/atlantmos/actions/runs/3844182193/jobs/6547064960#step:3:18)9,"caller":"atlantis-drift-detection/main.go:67","msg":"done call git clone ***github.com/jamengual/atlantmos.git /tmp/gogit2211482022","stdout":"","stderr":"Cloning into '/tmp/gogit2211482022'...\n"}
{"level":"panic","ts":1672897241.791465,"caller":"atlantis-drift-detection/main.go:143","msg":"failed to drift","error":"failed to parse repo config: error parsing config: repo config not allowed to set 'workflow' key: server-side config needs 'allowed_overrides: [workflow]'","stacktrace":"main.main\n\t/app/cmd/atlantis-drift-detection/main.go:143\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250"}
panic: failed to drift

goroutine 1 [running]:
go.uber.org/zap/zapcore.CheckWriteAction.OnWrite(0x1?, 0xc00012eea0?, {0x0?, 0x0?, 0xc000120260?})
	/go/pkg/mod/go.uber.org/zap@v1.22.0/zapcore/entry.go:198 +0x65
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc00012eb60, {0xc00009d280, 0x1, 0x1})
	/go/pkg/mod/go.uber.org/zap@v1.22.0/zapcore/entry.go:264 +0x3ec
go.uber.org/zap.(*Logger).Panic(0xc000[18](https://github.com/jamengual/atlantmos/actions/runs/3844182193/jobs/6547064960#step:3:19)4a[20](https://github.com/jamengual/atlantmos/actions/runs/3844182193/jobs/6547064960#step:3:21)?, {0xd40a10?, 0xc000046068?}, {0xc00009d280, 0x1, 0x1})
	/go/pkg/mod/go.uber.org/zap@v1.[22](https://github.com/jamengual/atlantmos/actions/runs/3844182193/jobs/6547064960#step:3:23).0/logger.go:[25](https://github.com/jamengual/atlantmos/actions/runs/3844182193/jobs/6547064960#step:3:26)1 +0x59
main.main()
	/app/cmd/atlantis-drift-detection/main.go:143 +0x12e5
```

And at this point there is 0 request to the atlantis server yet.

--- a/README.md
+++ b/README.md
@@ -8,13 +8,13 @@ The general workflow of this repository is:
 2. Find an atlantis.yaml file inside the repository
 3. Use atlantis to run /plan on each project in the atlantis.yaml file
 4. For each project with drift
-    1. Trigger a github workflow that can resolve the drift
-    2. Comment the existance of the drift in slack
-5. For each project directory in the atlantis.yamnl
+    1. Trigger a GitHub workflow that can resolve the drift
+    2. Comment the existence of the drift in slack
+5. For each project directory in the atlantis.yaml
    1. Run workspace list
    2. If any workspace isn't tracked by atlantis, notify slack
 
-There is an optional flag to cache drift results inside DynamoDB so we don't check the same directory twice in a short period of time.
+There is an optional flag to cache drift results inside DynamoDB, so we don't check the same directory twice in a short period of time.
 
 # Example for "Trigger a github workflow that can resolve the drift"
 

--- a/internal/atlantis/config.go
+++ b/internal/atlantis/config.go
@@ -4,9 +4,11 @@ import (
 	"fmt"
 	"github.com/runatlantis/atlantis/server/core/config"
 	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"gopkg.in/yaml.v3"
 	"os"
 	"path/filepath"
 	"sort"
+	"strings"
 )
 
 type DirectoriesWithWorkspaces map[string][]string
@@ -20,7 +22,7 @@ func (d DirectoriesWithWorkspaces) SortedKeys() []string {
 	return keys
 }
 
-func ConfigToWorkspaces(cfg *valid.RepoCfg) DirectoriesWithWorkspaces {
+func ConfigToWorkspaces(cfg *SimpleAtlantisConfig) DirectoriesWithWorkspaces {
 	workspaces := make(DirectoriesWithWorkspaces)
 	for _, p := range cfg.Projects {
 		if _, exists := workspaces[p.Dir]; !exists {
@@ -31,25 +33,20 @@ func ConfigToWorkspaces(cfg *valid.RepoCfg) DirectoriesWithWorkspaces {
 	return workspaces
 }
 
-func ParseRepoConfig(body string) (*valid.RepoCfg, error) {
-	t := true
-	var pv config.ParserValidator
-	vg := valid.GlobalCfg{
-		Repos: []valid.Repo{
-			{
-				ID:                   "terraform",
-				AllowCustomWorkflows: &t,
-			},
-		},
-	}
-	vc, err := pv.ParseRepoCfgData([]byte(body), vg, "terraform")
-	if err != nil {
-		return nil, fmt.Errorf("error parsing config: %s", err)
-	}
-	return &vc, nil
+type SimpleAtlantisConfig struct {
+	Version  int
+	Projects []valid.Project
 }
 
-func ParseRepoConfigFromDir(dir string) (*valid.RepoCfg, error) {
+func ParseRepoConfig(body string) (*SimpleAtlantisConfig, error) {
+	var ret SimpleAtlantisConfig
+	if err := yaml.NewDecoder(strings.NewReader(body)).Decode(&ret); err != nil {
+		return nil, fmt.Errorf("error parsing config: %s", err)
+	}
+	return &ret, nil
+}
+
+func ParseRepoConfigFromDir(dir string) (*SimpleAtlantisConfig, error) {
 	filename := filepath.Join(dir, config.AtlantisYAMLFilename)
 	body, err := os.ReadFile(filename)
 	if err != nil {

--- a/internal/atlantis/config_test.go
+++ b/internal/atlantis/config_test.go
@@ -26,6 +26,35 @@ projects:
     - '*.tf'
 `
 
+const exampleFromGithubIssue = `version: 3
+automerge: true
+delete_source_branch_on_merge: true
+parallel_plan: true
+parallel_apply: true
+allowed_regexp_prefixes:
+- lab/
+- staging/
+- prod/
+projects:
+- name: pepe-ue2-lab-cloudtrail
+  workspace: pepe-ue2-lab
+  workflow: workflow-1
+  dir: components/terraform/cloudtrail
+  terraform_version: v1.2.9
+  delete_source_branch_on_merge: false
+  autoplan:
+    enabled: true
+    when_modified:
+    - '**/*.tf'
+    - $PROJECT_NAME.tfvars.json
+  apply_requirements:
+  - approved`
+
+func TestParseRepoConfig(t *testing.T) {
+	_, err := ParseRepoConfig(exampleFromGithubIssue)
+	require.NoError(t, err)
+}
+
 func TestParseRepoConfigFromDir(t *testing.T) {
 	dirName, err := os.MkdirTemp("", "config-test")
 	require.NoError(t, err)

